kvbc/src/sparse_merkle/internal_node.cpp

```diff
@@ -386,12 +386,12 @@ size_t BatchedInternalNode::numLevel0Children() const {
   return std::count_if(children_.begin() + start_offset, children_.end(), [](const auto& c) { return c.has_value(); });
 }
 
-size_t BatchedInternalNode::nibbleToIndex(Nibble nibble) const {
+size_t BatchedInternalNode::nibbleToIndex(Nibble nibble) {
   constexpr size_t level_0_start = MAX_CHILDREN / 2;
   return level_0_start + nibble.data();
 }
 
-size_t BatchedInternalNode::height(size_t index) const {
+size_t BatchedInternalNode::height(size_t index) {
```

Comment:
Is it possible to document this strange function ?

Reply:
It is documented in the .h file, basically we pass the index of a child from the array **children_**, containing the 31 elements, and the return value is the height in the sub-tree. [internal_node.h:328](https://github.com/vmware/concord-bft/blob/66b249a1e709ed4d1baf5f4e5e73bae19bbadd58/kvbc/include/sparse_merkle/internal_node.h#L328)

Reply:
It is one of the functions that I changed to static, because they define the structure of the batched internal node and don't need to access the state of a particular instance.